### PR TITLE
Crash in g_print() on a certain pipeline, like x264enc ! mp4mux

### PR DESCRIPTION
**Pull request by singalen**

### CODE REVIEW

File: src/http-launch.c [hunk 1]
```diff
@@ -303,13 +303,19 @@ on_message (GstBus * bus, GstMessage * message, gpointer user_data)
...
-      gchar *debug;
-      GError *err;
+      gchar *debug = NULL;
+      GError *err = NULL;
 
       gst_message_parse_error (message, &err, &debug);
-      g_print ("Warning %s\n", err->message);
-      g_error_free (err);
-      g_free (debug);
+      //g_print ("Warning %s\n", err->message);
```
Comment (sdroege) [inline src/http-launch.c]: The problem here is that err is NULL? That shouldn't happen at all and would be a bug somewhere in GStreamer.

Can you get me a debug log with GST_DEBUG=6 of the crash in question?

Comment (singalen) [inline src/http-launch.c]: Indeed this seemed weird to me. I thought I don't know enough about GStreamer.
Here are my command and logs: https://gist.github.com/singalen/dd9005709b41a16bafd0
Ubuntu 14.10 x86_64, GStreamer 1.4.3.

Comment (sdroege) [inline src/http-launch.c]: Oops... the problem is that we should use gst_message_parse_warning() here, not gst_message_parse_error(). Copy and paste mistake.

Want to update the pull request?

Comment (singalen) [inline src/http-launch.c]: With a pleasure.

**Pull request closed without merging**